src/core/namespace.h

```diff
@@ -76,6 +77,7 @@ int setup_namespace(
                 const char *root_directory,
                 const char *root_image,
                 const MountOptions *root_image_options,
+                const MountEntry *mount_images,
```

Comment:
i am still not too happy that this exposes the same MountEntry structure. I'd really prefer if this was done like the `BindMount` stuff, i.e. an independent structure in an array. Yes, this means we'll copy things around a bit, but logically it keeps things nicely separate, and doesn't create the impression or fire the desire that we open up all kinds of mounts now to unit files.

Reply:
Ok, I'll change this to a separate structure.
I gather you prefer this version of the string parsing - with the changes to the handling of C escapes - rather than the nested EXTRACT_CUNESCAPE_RELAX? If so I'll cherry pick Zbigniew's commit

Reply:
Yes, I like @keszybz's work here very much

Reply:
ack - will cherry-pick and update https://github.com/systemd/systemd/pull/16321 with both requested changes

Reply:
pushed to https://github.com/systemd/systemd/pull/16321